### ISSUE TITLE
$USER / $SLURM_JOB_USER sometimes set to "nobody"

### ISSUE THREAD

**Issue by nhurleywalker** [nhurleywalker/GLEAM-X-pipeline#8]
If the LDAP server falls over, $USER is set to "nobody". If the SLURM server falls over, $SLURM_JOB_USER is set to "nobody". The first issue caused me to start using the second variable, but that still fails occasionally. This causes the GLEAM-X-pipeline.profile not to beloaded and a lot of semi-hard-coded locations to fall over. So we need to add something at the start of every script that tests each variable and uses whichever one is _not_ set to "nobody".

This issue can also be resolved by containerising absolutely everything -- although the container still has to be loaded by someone!

Tim, assigning this to you. I suggest doing a hasty hack to test the $USER and $SLURM_JOB_USER variables, and long-term, replace the profile loads and hardcodes with a better solution (maybe containers).

**Comment by tjgalvin** [comment]
I am looking at this now. When does the dodgy LDAP lookup that results in an `nobody` $USER occur? Does this only happen when a SLURM script is being executed on a worker node?

I think an alternative way to handle this would be to have the username inserted alongside some of the other parameters as the SLURM scripts are generated i.e. as a part of the set of `sed` expressions. This would not require any extra CLI arguments as it can be done transparently with the results of a `whoami` within each of the appropriate `obs_*` scripts. Everyone logged in and capable of running the pipeline on the current pawsey set up should have a `whoami` that does not return `nobody`- I would think?

Long term I think that there would be ways of wrapping up the required parameters and removing some of these hard-coded locations.

**Comment by nhurleywalker** [comment]
That's a good idea! It would also make it clear who ran the script. Please go ahead and implement that change if you get a chance.

**Comment by tjgalvin** [comment]
I have created a #11 that aims to correct this. In short, instances of `$USER` and `$SLURM_JOB_USER` have been replaced with a `PIPEUSER` string, which is used by the `sed` expressions to paste in a user name. This user name is based on the results of a `whoami` executed on the login node. Thinking about it, the `$USER` environment variable here should be ok, and could reasonably by used in place of a `whoami`. Hindsight... 

In any case, I have not exhaustively tested all scripts, so let me know how it all goes. I think it should just work. If you see any issues feel free to reject and let me know.
